### PR TITLE
tdesktop: 2.8.4 -> 2.8.11

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix [hunk 1]
```diff
@@ -64,6 +64,7 @@ in mkDerivation rec {
     "-DTDESKTOP_API_HASH=d524b414d21f4d37f08684c1df41ac9c"
     #"-DDESKTOP_APP_SPECIAL_TARGET=\"\"" # TODO: Error when set to "": Bad special target '""'
     "-DTDESKTOP_LAUNCHER_BASENAME=telegramdesktop" # Note: This is the default
+    "-DDESKTOP_APP_DISABLE_GTK_INTEGRATION=ON"
```
Comment (NickCao) [inline pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix:67]: We are having trouble building with GTK_INTEGRATION on, namely `error: template with C linkage`, anyway what is GTK integration?

**Comment by NickCao** [comment]
We might also be missing a few dependencies, `microsoft_gsl` and `rlottie`, but it builds without them.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130525` at 6fca6de3 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package skipped due to time constraints:</summary>
  <ul>
    <li>tdesktop</li>
  </ul>
</details>
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>libtgvoip</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix:109:5:
```
    |
109 |     license = licenses.gpl3;
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

util-linux is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix:49:3:
```
   |
49 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

extra-cmake-modules is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/networking/instant-messengers/telegram/tdesktop/default.nix:49:3:
```
   |
49 |   buildInputs = [
   |   ^
```
</li>
  </ul>
</details>

**Comment by ilya-fedin** [comment]
> We might also be missing a few dependencies, `microsoft_gsl` and `rlottie`, but it builds without them.

tdesktop has patched bundled rlottie, so it's not a good idea to package upstream rlottie and use as build input for telegram

**Review by primeos — APPROVED** [review]
Thanks! The diff LGTM and I double-checked the basic functionality.

**Comment by primeos** [comment]
@NickCao would you by any chance also be interested to become a maintainer of the `tdesktop` package?

**Comment by NickCao** [comment]
My answer would be positive.
> @NickCao would you by any chance also be interested to become a maintainer of the `tdesktop` package?

**Comment by primeos** [comment]
@NickCao awesome :) Feel free to add yourself as maintainer during the next update or to open a dedicated PR for it.

**Pull request merged**